client/my-sites/domain-tip/index.jsx

```diff
@@ -33,7 +33,10 @@ const DomainTip = React.createClass( {
 	propTypes: {
 		className: React.PropTypes.string,
 		event: React.PropTypes.string.isRequired,
-		siteId: React.PropTypes.number.isRequired
+		siteId: React.PropTypes.oneOfType( [
+			React.PropTypes.number,
+			React.PropTypes.bool
+		] )
```

Comment:
Maybe let's keep this as
```
siteId: React.PropTypes.number
```

and in mysites/stats/index we can update the DomainTip usage to:
```javascript
<DomainTip siteId={ site ? site.ID : null } event="stats_insights_domain" />
```